dipy/reconst/ivim.py

```diff
@@ -0,0 +1,540 @@
+            from None. default : ([0., 0., 0., 0.], [np.inf, .3, 1., 1.])
+
+        tol : float, optional
+            Tolerance for convergence of minimization.
+            default : 1e-15
+
+        f_threshold : float, optional
+            Threshold value to consider for f being erroneous after leastsq
+            fitting. If the value of f obtained crosses this threshold the
+            parameters will be taken from the linear fits.
+            default : 0.25
+
+        x_scale : array, optional
+            Scaling for the parameters. This is passed to `least_squares` which is
+            only available for Scipy version > 0.17.
+            default: [1000, 0.01, 0.001, 0.0001]
+
+        options : dict, optional
+            Dictionary containing gtol, ftol, eps and maxiter. This is passed
+            to leastsq.
+            default : options={'gtol': 1e-15, 'ftol': 1e-15, 'eps': 1e-15,
+                      'maxiter': 1000}
+
+        References
+        ----------
+        .. [1] Le Bihan, Denis, et al. "Separation of diffusion and perfusion
+               in intravoxel incoherent motion MR imaging." Radiology 168.2
+               (1988): 497-505.
+        .. [2] Federau, Christian, et al. "Quantitative measurement of brain
+               perfusion with intravoxel incoherent motion MR imaging."
+               Radiology 265.3 (2012): 874-881.
+        """
+        if not np.any(gtab.b0s_mask):
+            e_s = "No measured signal at bvalue == 0."
+            e_s += "The IVIM model requires signal measured at 0 bvalue"
+            raise ValueError(e_s)
+
+        ReconstModel.__init__(self, gtab)
+        self.split_b_D = split_b_D
+        self.split_b_S0 = split_b_S0
+        self.bounds = bounds
+        self.tol = tol
+        self.f_threshold = f_threshold
+        self.options = options
+        self.x_scale = x_scale
+
+        if SCIPY_LESS_0_17 and self.bounds is not None:
+            e_s = "Scipy versions less than 0.17 do not support "
+            e_s += "bounds. Please update to Scipy 0.17 to use bounds"
+            raise ValueError(e_s)
+        elif self.bounds is None:
+            self.bounds = ((0., 0., 0., 0.), (np.inf, 1., 1., 1.))
+        else:
+            self.bounds = bounds
+
+    @multi_voxel_fit
+    def fit(self, data, mask=None):
+        """ Fit method of the Ivim model class.
+
+        The fitting takes place in the following steps: Linear fitting for D
+        (bvals > 200) and store S0_prime. Another linear fit for S0 (bvals <
+        200).Estimate f using 1 - S0_prime/S0. Use least squares to fit D_star
+        and f.
+
+        We do a final fitting of all four parameters and select the set of
+        parameters which make sense physically. The criteria for selecting a
+        particular set of parameters is checking the perfusion fraction.
+        If the fraction is more than 25%, we will reject the solution obtained
```

Comment:
Sorry for being a pain in the neck about this. Given that it is an argument it will be 25% *per default*, but will change based on user input. Can we write: 
   
    ... is more than `f_threshold` (default: 25%), we will reject...